### PR TITLE
Record Layer Updates

### PR DESCRIPTION
**Pull request by hannestschofenig**

### CODE REVIEW

File: draft-ietf-tls-ctls.md [hunk 1]
```diff
@@ -130,37 +130,69 @@ affected.  We do not show the structures which only change in this way.
...
 
-The cTLS Record Layer assumes that records are externally framed
-(i.e., that the length is already known because it is carried in a UDP
-datagram or the like). Depending on how this was carried, you might
-need another byte or two for that framing. Thus, only the type byte
-need be carried and TLSPlaintext becomes:
+The only cTLS records that are sent in plaintext are handshake records
+(ClientHello and ServerHello/HRR).  The content type is therefore constant (it
+is always `handshake`), so we instead set the `content_type` field to a fixed
+cTLS-specific valud to distinguish cTLS plaintext records from encrypted
```
Comment (ekr) [inline draft-ietf-tls-ctls.md:136]: Why can't we just use handshake here? It's distinguishable from the encrypted DTLS content types.

File: draft-ietf-tls-ctls.md [hunk 2]
```diff
@@ -130,37 +130,69 @@ affected.  We do not show the structures which only change in this way.
...
+if no compression profile is used.  Non-zero values are negotiated out of band
+between the client and server, as part of the specification of the compression
+profile.
 
 ~~~~
       struct {
-          ContentType type;
-          opaque fragment[TLSPlaintext.length];
-      } TLSPlaintext;
+          ContentType content_type = ctls;
```
Comment (ekr) [inline draft-ietf-tls-ctls.md:148]:
```suggestion
          ContentType content_type = ctls_handshake;
```

File: draft-ietf-tls-ctls.md [hunk 3]
```diff
@@ -130,37 +130,69 @@ affected.  We do not show the structures which only change in this way.
...
+          varint profile_id;
```
Comment (ekr) [inline draft-ietf-tls-ctls.md:149]: Do these have to match? What comes from the server?

File: draft-ietf-tls-ctls.md [hunk 4]
```diff
@@ -512,8 +554,16 @@ which profile.
...
+This document requests that a code point be allocated from the "TLS ContentType
+registry.  This value must be in the range 0-31 (inclusive).  The row to be
+added in the registry has the following form:
+
+| Value | Description | DTLS-OK | Reference |
+|:=====:|:============|:========|:==========|
+|  TBD  | ctls        | N       | RFCXXXX   |
 
+[[ RFC EDITOR: Please replace the value TBD with the value assigned by IANA, and
+the value XXXX to the RFC number assigned for this document. ]]
```
Comment (ekr) [inline draft-ietf-tls-ctls.md:566]: I suggest that we require standards action for all profile IDs that would fit in 2 octets.

File: draft-ietf-tls-ctls.md (hunk 1 above)
Comment (ekr) [reply draft-ietf-tls-ctls.md:136]: nm. I see why.

File: draft-ietf-tls-ctls.md (hunk 3 above)
Comment (hannestschofenig) [reply draft-ietf-tls-ctls.md:149]: We would only need a profile ID transmitted in the ClientHello. As such, I wonder whether we should omit the profile_id for ServerHellos?

Comment (ekr) [reply draft-ietf-tls-ctls.md:149]: Maybe an open issue marker?

Comment (hannestschofenig) [reply draft-ietf-tls-ctls.md:149]: Done.

**Pull request merged**